app/src/main/java/org/oppia/app/utility/ClickableAreasImage.kt

```diff
@@ -0,0 +1,112 @@
+package org.oppia.app.utility
+
+import android.graphics.RectF
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.ImageView
+import androidx.core.view.forEachIndexed
+import androidx.core.view.isVisible
+import com.github.chrisbanes.photoview.OnPhotoTapListener
+import com.github.chrisbanes.photoview.PhotoViewAttacher
+import org.oppia.app.R
+import org.oppia.app.player.state.ImageRegionSelectionInteractionView
+import kotlin.math.roundToInt
+
+class ClickableAreasImage(
+  private val imageView: ImageRegionSelectionInteractionView,
+  private val parentView: FrameLayout,
+  private val listener: OnClickableAreaClickedListener
+) : OnPhotoTapListener {
+  private val attacher: PhotoViewAttacher = PhotoViewAttacher(imageView)
+
+  init {
+    attacher.setOnPhotoTapListener(this)
+  }
+
+  /**
+   * Called when an image is clicked.
+   *
+   * @param view the original view on which the tap/click occurs.
+   * @param x the relative x coordinate according to image
+   * @param y the relative y coordinate according to image
+   */
+  override fun onPhotoTap(view: ImageView, x: Float, y: Float) {
+    val clickableAreaIndex = getClickableAreaOrDefault(x, y)
+    parentView.forEachIndexed { index: Int, tappedView: View ->
+      if (index > 0) {
+        tappedView.isVisible = index == clickableAreaIndex + 1
+        listener.onClickableAreaTouched(
+          imageView.getClickableAreas()[clickableAreaIndex].label
+        )
+      }
+    }
+  }
+
+  private fun getClickableAreaOrDefault(x: Float, y: Float): Int {
+    return imageView.getClickableAreas().indexOfFirst {
+      isBetween(it.region.area.upperLeft.x, it.region.area.lowerRight.x, x) &&
+        isBetween(it.region.area.upperLeft.y, it.region.area.lowerRight.y, y)
+    }
+  }
+
+  /** Return whether a point lies between two points.*/
+  private fun isBetween(start: Float, end: Float, actual: Float): Boolean {
+    return actual in start..end
+  }
+
+  /** Get X co-ordinate scaled according to image.*/
+  private fun getXCoordinate(x: Float): Float {
+    val rect = attacher.displayRect
+    return (x * rect.width()) + rect.left
+  }
+
+  /** Get Y co-ordinate scaled according to image.*/
+  private fun getYCoordinate(y: Float): Float {
+    val rect = attacher.displayRect
+    return (y * rect.height()) + rect.top
+  }
+
+  fun addViews(isAccessible: Boolean) {
```

Comment:
Instead of ``isAccessible``, we should describe this property based on what it's actually providing, e.g. ``useSeparateRegionViews`` or similar.